baybe/objectives/single.py

```diff
@@ -0,0 +1,26 @@
+"""Functionality for single-target objectives."""
+
+import pandas as pd
+from attr import define, field
+from attr.validators import instance_of
+
+from baybe.objectives.base import Objective
+from baybe.targets.base import Target
+
+
+@define(frozen=True)
+class SingleTargetObjective(Objective):
+    """An objective focusing on a single target."""
+
+    _target: Target = field(validator=instance_of(Target))  # type: ignore[type-abstract]
+    """The single target considered by the objective."""
+
+    @property
+    def targets(self) -> tuple[Target, ...]:  # noqa: D102
```

Comment:
Please not that this one here implements the abstract property from the base class. So this must be plural here. I the plural is what we should expose in the base class since other modules can then easily work with all sorts of objectives.

Related: there is a new "override" decorator in python that has exactly the purpose of indicating this relationship (and avoiding that you create a new property be mistake, e.g. because of a spelling mistake). Do you think we should add this to our codebase? Would be a separate PR though, which would introduce it everywhere.